src/assets/js/scope.js

```diff
@@ -0,0 +1,18 @@
+class Scope {
```

Comment:
Can we add some documentation here to what this file is for? The name is scope, but that term is so overloaded that I don't directly see that it's actually a permissions based handler. I think the ```scope``` in this place is a pretty generic term that doesn't actually tell me what it is, but it's already all over the code, so maybe at least some documentation of what this is meant for so future developers don't try to put stuff in here that doesn't belong here would be great.